src/postgres/commands/checkAuthentication.ts

```diff
@@ -24,10 +29,21 @@ export async function checkAuthentication(context: IActionContext, treeItem: Pos
                 // Need to configure firewall only for Azure Subscritption accounts
             } else if (treeItem.parent.resourceGroup && parsedError.errorType === firewallNotConfiguredErrorType) {
                 await configurePostgresFirewall(context, treeItem.parent);
+            } else if (treeItem.parent.resourceGroup && parsedError.errorType === 'ETIMEDOUT' && !(await isFirewallRuleSet(treeItem.parent))) {
+                await configurePostgresFirewall(context, treeItem.parent);
             } else {
                 throw error;
             }
         }
     }
     return clientConfig;
 }
+
+export async function isFirewallRuleSet(treeItem: PostgresServerTreeItem): Promise<boolean> {
+    const serverType: PostgresServerType = nonNullProp(treeItem, 'serverType');
+    const client = createAbstractPostgresClient(serverType, treeItem.root);
+    client.firewallRules.listByServer
+    const result: FirewallRuleListResult = (await client.firewallRules.listByServer(nonNullProp(treeItem, 'resourceGroup'), nonNullProp(treeItem, 'azureName')))._response.parsedBody;
+    return (result.some(async value => value.name === 'azureDatabasesForVSCode-publicIp' && value.startIpAddress === await getPublicIp()));
```

Comment:
I don't think we need to check the name, right? This logic should work regardless of where the firewall rule is set (in our extension or outside of it)

Reply:
That's true, the name should be irrelevant, especially if a user sets the rule on the portal.